routing/payment_lifecycle.go

```diff
@@ -286,6 +417,49 @@ type shardResult struct {
 	err error
 }
 
+// collectResultAsync launches a goroutine that will wait for the result of the
+// given HTLC attempt to be available then handle its result. Note that it will
+// fail the payment with the control tower if a terminal error is encountered.
+func (p *shardHandler) collectResultAsync(attempt *channeldb.HTLCAttemptInfo) {
+
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+
+		// Block until the result is available.
+		result, err := p.collectResult(attempt)
+		if err != nil {
+			log.Errorf("Error collecting result for shard %v for "+
+				"payment %v: %v", attempt.AttemptID,
+				p.paymentHash, err)
+
+			select {
+			case p.shardErrors <- err:
+			case <-p.quit:
```

Comment:
Wait also for `router.quit`? Preferably we have a solution with only a single quit channel. Some for two instances below.

Reply:
Added router quit channel.